gpflow/models/sgpr.py

```diff
@@ -150,46 +151,95 @@ class SGPR(SGPRBase):
 
     """
 
+    CommonTensors = namedtuple("CommonTensors", ["A", "B", "LB", "AAT", "L"])
+
+
     def maximum_log_likelihood_objective(self) -> tf.Tensor:
         return self.elbo()
-
-    def elbo(self) -> tf.Tensor:
+    
+    def _common_calculation(self):
         """
-        Construct a tensorflow function to compute the bound on the marginal
-        likelihood. For a derivation of the terms in here, see the associated
-        SGPR notebook.
+        Matrices used in log-det calculation
+        :return: A, B, LB, AAT all square NxN matrices
         """
-        X_data, Y_data = self.data
-
-        num_inducing = self.inducing_variable.num_inducing
-        num_data = to_default_float(tf.shape(Y_data)[0])
-        output_dim = to_default_float(tf.shape(Y_data)[1])
+        x, y = self.data
+        iv = self.inducing_variable
+        num_iv = tf.shape(iv.Z)[0]
+        sigma_sq = self.likelihood.variance
 
-        err = Y_data - self.mean_function(X_data)
-        Kdiag = self.kernel(X_data, full_cov=False)
-        kuf = Kuf(self.inducing_variable, self.kernel, X_data)
-        kuu = Kuu(self.inducing_variable, self.kernel, jitter=default_jitter())
+        kuf = Kuf(iv, self.kernel, x)
+        kuu = Kuu(iv, self.kernel, jitter=default_jitter())
         L = tf.linalg.cholesky(kuu)
-        sigma = tf.sqrt(self.likelihood.variance)
+        sigma = tf.sqrt(sigma_sq)
 
         # Compute intermediate matrices
         A = tf.linalg.triangular_solve(L, kuf, lower=True) / sigma
         AAT = tf.linalg.matmul(A, A, transpose_b=True)
-        B = AAT + tf.eye(num_inducing, dtype=default_float())
+        B = AAT + tf.eye(num_iv, dtype=default_float())
```

Comment:
Using `set_diag` to avoid forming an identity matrix and adding explicitly would be preferable.